flink-runtime/src/main/java/org/apache/flink/runtime/io/compression/AirDecompressor.java

```diff
@@ -0,0 +1,102 @@
+ */
+
+package org.apache.flink.runtime.io.compression;
+
+import io.airlift.compress.Decompressor;
+import io.airlift.compress.MalformedInputException;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import static org.apache.flink.runtime.io.compression.CompressorUtils.readIntLE;
+import static org.apache.flink.runtime.io.compression.CompressorUtils.validateLength;
+import static org.apache.flink.runtime.io.compression.Lz4BlockCompressionFactory.HEADER_LENGTH;
+
+/** Flink decompressor that wrap {@link io.airlift.compress.Decompressor}. */
+public class AirDecompressor implements BlockDecompressor {
+    Decompressor internalDecompressor;
+
+    public AirDecompressor(Decompressor internalDecompressor) {
+        this.internalDecompressor = internalDecompressor;
+    }
+
+    @Override
+    public int decompress(ByteBuffer src, int srcOff, int srcLen, ByteBuffer dst, int dstOff)
+            throws DataCorruptionException, InsufficientBufferException {
+        final int prevSrcOff = src.position() + srcOff;
+        final int prevDstOff = dst.position() + dstOff;
+
+        src.position(prevSrcOff);
+        dst.position(prevDstOff);
+        src.order(ByteOrder.LITTLE_ENDIAN);
+        final int compressedLen = src.getInt();
+        final int originalLen = src.getInt();
+        validateLength(compressedLen, originalLen);
+
+        if (dst.capacity() - prevDstOff < originalLen) {
+            throw new InsufficientBufferException("Buffer length too small");
+        }
+
+        if (src.limit() - prevSrcOff - HEADER_LENGTH < compressedLen) {
+            throw new DataCorruptionException("Source data is not integral for decompression.");
+        }
+        src.limit(prevSrcOff + compressedLen + HEADER_LENGTH);
+        try {
+            internalDecompressor.decompress(src, dst);
+            int originalLen2 = dst.position() - prevDstOff;
+            if (originalLen != originalLen2) {
+                throw new DataCorruptionException(
+                        "Input is corrupted, unexpected original length.");
+            }
+        } catch (MalformedInputException e) {
+            throw new DataCorruptionException("Input is corrupted", e);
+        }
+
+        return originalLen;
+    }
+
+    @Override
+    public int decompress(byte[] src, int srcOff, int srcLen, byte[] dst, int dstOff)
+            throws DataCorruptionException, InsufficientBufferException {
+        int compressedLen = readIntLE(src, srcOff);
+        int originalLen = readIntLE(src, srcOff + 4);
+        validateLength(compressedLen, originalLen);
+
+        if (dst.length - dstOff < originalLen) {
+            throw new InsufficientBufferException("Buffer length too small");
+        }
+
+        if (src.length - srcOff - HEADER_LENGTH < compressedLen) {
+            throw new DataCorruptionException("Source data is not integral for decompression.");
+        }
+
+        try {
+            final int originalLen2 =
```

Comment:
nit: may rename originalLen2  to decompressedLen

Reply:
Renamed.